src/models/tools/geometry/geometry-content.test.ts

```diff
@@ -125,4 +133,76 @@ describe("GeometryContent", () => {
     polygon = content.applyChange(board, { operation: "create", target: "polygon" }) as JXG.Polygon;
     expect(polygon).toBeUndefined();
   });
+
+  it("can add an image", () => {
+    const content = defaultGeometryContent();
+    const board = createDefaultBoard(content);
+    const image = content.addImage(board, placeholderImage, [0, 0], [5, 5]);
+    expect(image!.elType).toBe("image");
+    content.updateObjects(board, image!.id, { url: placeholderImage });
+    expect(image!.url).toBe(placeholderImage);
+    content.updateObjects(board, image!.id, { size: [10, 10] });
+    content.updateObjects(board, image!.id, { url: placeholderImage, size: [10, 10] });
+    expect(image!.url).toBe(placeholderImage);
+
+    const change: JXGChange = {
+      operation: "create",
+      target: "image",
+      parents: [placeholderImage],
+      properties: { id: "image-fail" }
+    };
+    const failedImage = content.applyChange(board, change);
+    expect(failedImage).toBeUndefined();
+  });
+
+  it("can select points, etc.", () => {
+    const content = defaultGeometryContent();
+    const metadata = GeometryMetadataModel.create({ id: "geometry-1" });
+    content.doPostCreate(metadata);
+    const board = createDefaultBoard(content);
+    const p1 = content.addPoint(board, [0, 0]);
+    const p2 = content.addPoint(board, [1, 1]);
+    const p3 = content.addPoint(board, [1, 0]);
+    const poly = content.createPolygonFromFreePoints(board);
+    content.selectObjects(board, p1!.id);
+    expect(content.isSelected(p1!.id)).toBe(true);
+    expect(content.isSelected(p2!.id)).toBe(false);
+    content.selectObjects(board, poly!.id);
+    expect(content.isSelected(poly!.id)).toBe(true);
```

Comment:
Would we also expect each of the individual points to be selected, or just the polygon?

Reply:
No, the logic to of when/whether to select children is in GeometryToolComponent not in the model, so this wouldn't automatically select the points. It's something to consider, however if we think selecting ancestors (points are ancestors of polygons in JSXGraph) is always desirable.